lib/config/npmrc.ts

```diff
@@ -0,0 +1,64 @@
+import ini from 'ini';
+import { id as hostType } from '../datasource/npm';
+import { RenovateConfig } from './types';
+
+function getMatchHost(input: string): string {
+  if (input.startsWith('https://')) {
+    return input;
+  }
+  return input.split('//')[1];
+}
+
+export function getConfigFromNpmrc(npmrc = ''): RenovateConfig {
+  const res: RenovateConfig = {};
+  const parsed = ini.parse(npmrc);
+  for (const [key, val] of Object.entries(parsed)) {
+    // hostRules
+    if (key === '_auth') {
+      res.hostRules ||= [];
+      res.hostRules.push({
+        hostType,
+        authType: 'Basic',
+        token: val,
+      });
+    } else if (key === '_authToken') {
+      res.hostRules ||= [];
+      res.hostRules.push({ hostType, token: val });
+    } else if (key.endsWith(':_auth')) {
+      const hostPart = key.replace(/:_auth$/, '');
+      res.hostRules ||= [];
+      res.hostRules.push({
+        hostType,
+        matchHost: getMatchHost(hostPart),
+        authType: 'Basic',
+        token: val,
+      });
+    } else if (key.endsWith(':_authToken')) {
+      const hostPart = key.replace(/:_authToken$/, '');
+      res.hostRules ||= [];
+      res.hostRules.push({
+        hostType,
+        matchHost: getMatchHost(hostPart),
```

Comment:
Hostpart always starts with `//` for npmrc. `http` or `https` is defined by registry url setting.